crypto/evp/pmeth_lib.c

```diff
@@ -1337,6 +1337,14 @@ static int legacy_ctrl_to_param(EVP_PKEY_CTX *ctx, int keytype, int optype,
             return EVP_PKEY_CTX_set_rsa_keygen_primes(ctx, p1);
         }
     }
+
+    if (keytype == EVP_PKEY_RSA_PSS) {
+      switch(cmd) {
+        case EVP_PKEY_CTRL_MD:
+          return ctx->pmeth->ctrl(ctx, cmd, p1, p2);
```

Comment:
Why this direct call instead of `EVP_PKEY_CTX_set_rsa_pss_keygen_md(ctx, p2)`?

Reply:
This is incorrect. This function is called `legacy_ctrl_to_param`. Its purpose is to convert instances where the user has made legacy ctrl calls, and they need to be converted into new style params calls. This is just calling the legacy ctrl instead.

Reply:
Sorry about that, I'll take a look now. Thanks